client/src/lib/src/utils/files.ts

```diff
@@ -5,7 +5,7 @@
 
 import path from 'path'
 
-export function extractRecipeName (filename: string | undefined): string | undefined {
-  if (filename === undefined) { return undefined }
-  return path.basename(filename).split('.')[0].split('_')[0]
+export function extractRecipeName (filePath: string | undefined): string | undefined {
+  if (filePath === undefined) { return undefined }
+  return path.parse(filePath).name.split('_')[0]
```

Comment:
@deribaucourt do we need to consider the cases like `path/recipe_name_1.0.bb` where multiple underscores are present?

Reply:
recipe names can only have one underscore, the rest is considered to be the version. In your case, the recipe name is `recipe`, and the version `name_1.0`

Reply:
Use `extractRecipeName` to remove code duplication

Reply:
where is the duplication?